hs628_tools/PCA_refinement.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Jul 21 14:43:54 2018
+
+@author: BlauBear
+"""
+
+import numpy as np
+import pandas as pd
+from sklearn.cross_validation import cross_val_score, ShuffleSplit
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.preprocessing import StandardScaler
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.pipeline import Pipeline
+from sklearn.decomposition import PCA
+
+#find out which variables are correlated
+#X is the independent variables, Y is the dependent variables, names is the variable names
+def correlated_variables (X, Y, names):
+    rf = RandomForestRegressor(n_estimators=20, max_depth=4)
+    scores = []
+    for i in range(X.shape[1]):
```

Comment:
Hi Choi,
I am not sure why you chose to loop through the shape of independent variable X. Is there any particular reason why you took this approach?
Over all code looks good except for missing docstrings. I believe you could use some sentences to explain what those user defined functions do and how you plan on executing the given function.

Good job!!

--Nirvesh

Reply:
Because my intention is as it loops through the independent variables, it checks for correlation between the one which is being looped through and the other independent variables, like in line 24. So it returns both the variables and the scores of the correlation.

Reply:
If you refer to a line simply as "line 24" then a later commit can easily interfere with your reference-- in this case I don't think that the current line 24 was what you really intended to refer to.  To safely make a reference to a particular line, follow [the procedure described here](https://stackoverflow.com/a/31282559).